src/map/entities/charentity.cpp

```diff
@@ -1212,6 +1211,13 @@ void CCharEntity::OnCastFinished(CMagicState& state, action_t& action)
                         actionTarget.addEffectMessage = 287 + effect;
                     }
                     actionTarget.additionalEffect = effect;
+
+                    // Closing a skillchain with an immanence Helix will make the magic burst window longer
+                    auto scEffect = PTarget->StatusEffectContainer->GetStatusEffect(EFFECT_SKILLCHAIN, 0);
+                    if (isHelix && scEffect)
+                    {
+                        scEffect->SetDuration(scEffect->GetDuration() + 2000);
```

Comment:
> [WinterSolstice8](https://github.com/WinterSolstice8) [2 weeks ago](https://github.com/LandSandBoat/server/pull/5525#discussion_r1581879360) • 
Second thought: should this not get the duration then add 2000? Skillchains are supposed to have reduced duration at least for following skillchains.

Reply:
Is this a completely flat 2s bonus, or does it scale with stats/gear/something else? I know _nothing_ about SCH, and this isn't listed obviously when I google `ffxi immanence`

Reply:
Yep. Had it confirmed when I made the PR

https://www.bg-wiki.com/index.php?title=Immanence&diff=prev&oldid=698295

Reply:
Delightful, thanks 👍